src/www/ui_components/server_list/types.ts

```diff
@@ -0,0 +1,11 @@
+import {ServerConnectionState} from "../server_card";
```

Comment:
Isn't this creating a circular dependency?

Also, this type is part of the server list interface. Could you explain the motivation for the refactoring?